### PR TITLE
api: omit empty fields in /experiments/detail

### PR DESCRIPTION
**Pull request by g1eny0ung**
### What problem does this PR solve?

This PR omits the empty fields in the `/experiments/detail` response.

### What is changed and how does it work?

For example, a NetworkChaos:

> Before:

```json
"target": {
    "kind": "",
    "pod_chaos": {
      "action": "",
      "container_name": ""
    },
    "network_chaos": {
      "action": "delay",
      "delay": {
        "latency": "30ms",
        "correlation": "0",
        "jitter": "0ms"
      },
      "loss": null,
      "duplicate": null,
      "corrupt": null,
      "bandwidth": null,
      "direction": "to",
      "target_scope": {
        "namespace_selectors": [
          "default"
        ],
        "label_selectors": {
          "app": "hello-minikube"
        },
        "annotation_selectors": null,
        "field_selectors": null,
        "phase_selectors": null,
        "pods": null,
        "mode": "",
        "value": ""
      }
    },
    "io_chaos": {
      "action": "",
      "addr": "",
      "delay": "",
      "errno": "",
      "path": "",
      "percent": "",
      "methods": null
    },
    "kernel_chaos": {
      "fail_kernel_req": {
        "failtype": 0
      }
    },
    "time_chaos": {
      "offset": "",
      "clock_ids": null,
      "container_names": null
    },
    "stress_chaos": {
      "stressors": null
    }
  }
```

> After:

```json
"target": {
    "kind": "NetworkChaos",
    "network_chaos": {
      "action": "delay",
      "delay": {
        "latency": "30ms",
        "correlation": "0",
        "jitter": "0ms"
      },
      "loss": null,
      "duplicate": null,
      "corrupt": null,
      "bandwidth": null,
      "direction": "to",
      "target_scope": {
        "namespace_selectors": [
          "default"
        ],
        "label_selectors": {
          "app": "hello-minikube"
        },
        "annotation_selectors": null,
        "field_selectors": null,
        "phase_selectors": null,
        "pods": null,
        "mode": "",
        "value": ""
      }
    }
  }
```

### Check List

Code changes

- Has Go code change

### CODE REVIEW

**Comment by fewdan** [comment]
please fix CI

**Comment by g1eny0ung** [comment]
@fewdan Already fixed. Thx for mentioned!

File: pkg/apiserver/experiment/experiment.go [hunk 1]
```diff
@@ -164,13 +164,13 @@ func (s *SelectorInfo) ParseSelector() v1alpha1.SelectorSpec {
...
 type TargetInfo struct {
-	Kind         string           `json:"kind" binding:"required,oneof=PodChaos NetworkChaos IoChaos KernelChaos TimeChaos StressChaos"`
-	PodChaos     PodChaosInfo     `json:"pod_chaos"`
-	NetworkChaos NetworkChaosInfo `json:"network_chaos"`
-	IOChaos      IOChaosInfo      `json:"io_chaos"`
-	KernelChaos  KernelChaosInfo  `json:"kernel_chaos"`
-	TimeChaos    TimeChaosInfo    `json:"time_chaos"`
-	StressChaos  StressChaosInfo  `json:"stress_chaos"`
+	Kind         string            `json:"kind" binding:"required,oneof=PodChaos NetworkChaos IoChaos KernelChaos TimeChaos StressChaos"`
+	PodChaos     *PodChaosInfo     `json:"pod_chaos,omitempty"`
```
Comment (cwen0) [inline pkg/apiserver/experiment/experiment.go:168]: If we use the pointer here, we should check the `nil` when using them.

**Pull request merged**